src/app/clusters/push-av-stream-transport-server/push-av-stream-transport-logic.cpp

```diff
@@ -571,29 +605,36 @@ PushAvStreamTransportServerLogic::HandleAllocatePushTransport(CommandHandler & h
         return std::nullopt;
     }
 
-    IngestMethodsEnum ingestMethod = commandData.transportOptions.ingestMethod;
-
-    bool isFormatSupported = false;
-
-    for (auto & supportsFormat : mSupportedFormats)
+    // here add check for valid zoneid
+    if ((transportOptions.triggerOptions.triggerType == TransportTriggerTypeEnum::kMotion) &&
+        (transportOptions.triggerOptions.motionZones.HasValue()) && (!transportOptions.triggerOptions.motionZones.Value().IsNull()))
     {
-        if ((supportsFormat.ingestMethod == ingestMethod) &&
-            (supportsFormat.containerFormat == commandData.transportOptions.containerOptions.containerType))
+
+        auto & motionZonesList = transportOptions.triggerOptions.motionZones;
+        auto iter              = motionZonesList.Value().Value().begin();
+        uint16_t zonesize      = 0;
+        auto itsz              = motionZonesList.Value().Value().begin();
+        while (itsz.Next())
         {
-            isFormatSupported = true;
+            zonesize += 1;
         }
-    }
+        bool isValidZoneSize = mDelegate->ValidateMotionZoneSize(zonesize);
+        VerifyOrReturnValue(
+            isValidZoneSize, Status::ConstraintError,
+            ChipLogError(Zcl, "Transport Options verification from command data[ep=%d]: Invalid Motion Zone Size ", mEndpointId));
 
-    if (isFormatSupported == false)
-    {
-        auto status = to_underlying(StatusCodeEnum::kInvalidCombination);
-        ChipLogError(Zcl,
-                     "HandleAllocatePushTransport[ep=%d]: Invalid Ingest Method and Container Format Combination : (Ingest Method: "
-                     "%02X and Container Format: %02X)",
-                     mEndpointId, to_underlying(ingestMethod),
-                     to_underlying(commandData.transportOptions.containerOptions.containerType));
-        handler.AddClusterSpecificFailure(commandPath, status);
-        return std::nullopt;
+        while (iter.Next())
+        {
+            auto & transportZoneOption = iter.GetValue();
+            Status zoneIdStatus        = mDelegate->ValidateZoneId(transportZoneOption.zone.Value());
```

Comment:
Won't this crash if `Zone` is null?

Reply:
A Zone can be Null, it means triggering via Motion anywhere vs. in the defined ZoneID (per spec).